proposals/mmmm-spawn-let.md

```diff
@@ -0,0 +1,701 @@
+
+```swift
+func greet(_ f: @autoclosure () async -> String) async -> String { await f() }
+
+spawn let name = "Bob"
+await greet(await name) // await on name is required, because autoclosure
+```
+
+It is *not* legal to escape a `spawn let` value to an escaping closure. This is because structures backing the spawn let implementation may be allocated on the stack rather than the heap. This makes them very efficient, and makes great use of the structured guarantees they have to adhere to. These optimizations, however, make it unsafe to pass them to any escaping contexts:
+
+```swift
+func greet(_ f: @escaping () async -> String) async -> String { somewhere = f; somewhere() }
+
+spawn let name = "Bob"
+await greet(await name) // error: cannot escape 'spawn let' value
+```
+
+
+
+> Note: If Swift had a `@useImmediately` annotation that could be used together with even escaping closures, as they would "promise" to be called immediately without detaching or storing the closure elsewhere.
+
+### `spawn let` error propagation
+
+While it is legal to declare a `spawn let` and never explicitly `await` on it, it also implies that we do not particuilary care about its result.
+
+This is the same as spawning a number of child-tasks in a task group, and not collecting their results, like so:
+
+```swift
+try await withThrowingTaskGroup(of: Int.self) { group in 
+  group.spawn { throw Boom() }
+                             
+  return 0 // we didn't care about the child-task at all(!)
+} // returns 0
+```
+
+The above TaskGroup example will ignore the `Boom` thrown by its child task. However, it _will_ await for the task (and any other tasks it had spawned) to run to completion before the `withThrowingTaskGroup` returns. If we wanted to surface all potential throws of tasks spawned in the group, we should have written: `for try await _ in group {}` which would have re-thrown the `Boom()`.
+
+The same concept carries over to `spawn let`, where the scope of the group is replaced by the syntactic scope in which the `spawn let` was declared. For example, the following snippet is semantically equivalent to the above TaskGroup one:
+
+```swift
+// func boom() throws -> Int { throw Boom() }
+
+func work() async -> Int {
+  spawn let work: Int = boom()
+  // never await work...
+  return 0
+  // implicitly: cancels work
+  // implicitly: awaits work, discards errors
+}
+```
+
+This work function, will never throw, because we didn't await on the throwing `spawn let`. If we modified it to explicitly await on it, the compiler would force us to spell out not only the `await` but also the `try` keyword. The presence of the `try` keyword woult then force us to annotate the `work()` function as `throws`, as expected from normal, non-asynchronous code in Swift:
```

Comment:
`woult` → `would`

```suggestion
This `work()` function will never throw, because we didn't await on the throwing `spawn let`. If we modified it to explicitly await on it, the compiler would force us to spell out not only the `await` but also the `try` keyword. The presence of the `try` keyword would then force us to annotate the `work()` function as `throws`, as expected from normal, non-asynchronous code in Swift:
```